src/components/product/product.service.ts

```diff
@@ -63,63 +74,116 @@ export class ProductService {
+          active: true,
+          id: id,
+        }),
+      ])
+      .optionalMatch([...this.propMatch('type')])
+      .optionalMatch([...this.propMatch('books')])
+      .optionalMatch([...this.propMatch('mediums')])
+      .optionalMatch([...this.propMatch('purposes')])
+      .optionalMatch([...this.propMatch('approach')])
+      .optionalMatch([...this.propMatch('methodology')])
+      .return({
+        product: [{ id: 'id' }],
+        type: [{ value: 'type' }],
+        books: [{ value: 'books' }],
+        mediums: [{ value: 'mediums' }],
+        purposes: [{ value: 'purposes' }],
+        approach: [{ value: 'approach' }],
+        methodology: [{ value: 'methodology' }],
+      });
+
+    const result = await readProduct.first();
 
-    if (!result || !result.id.value) {
+    if (!result || !result.id) {
+      this.logger.warning(`Could not find product`, { id: id });
       throw new NotFoundException('Could not find product');
     }
 
     return {
       id,
-      createdAt: result.createdAt.value,
-      type: result.type.value,
-      books: result.books?.value || [],
-      mediums: result.mediums?.value || [],
-      purposes: result.purposes?.value || [],
-      approach: result.approach.value,
-      methodology: result.methodology.value,
+      createdAt: result.createdAt,
+      type: result.type,
+      books: result.books,
+      mediums: result.mediums,
+      purposes: result.purposes,
+      approach: result.approach,
+      methodology: result.methodology,
     };
   }
 
   async list(
     { page, count, sort, order, filter }: ProductListInput,
     session: ISession
   ): Promise<ProductListOutput> {
-    const result = await this.db.list<Product>({
-      session,
-      nodevar: 'product',
-      aclReadProp: 'canReadProducts',
-      aclEditProp: 'canCreateProduct',
-      props: [
-        { name: 'type', secure: false },
-        { name: 'books', secure: false, list: true },
-        { name: 'mediums', secure: false, list: true },
-        { name: 'purposes', secure: false, list: true },
-        { name: 'approach', secure: false },
-        { name: 'methodology', secure: false },
-      ],
-      input: {
-        page,
+    const { engagementId } = filter;
+    let result: {
+      items: Product[];
+      hasMore: boolean;
+      total: number;
+    } = { items: [], hasMore: false, total: 0 };
+
+    if (engagementId) {
+      const qry = `
+        MATCH
+          (token:Token {active: true, value: $token})
+          <-[:token {active: true}]-
+          (requestingUser:User {
+            active: true,
+            id: $requestingUserId
+          }),
+          (engagement {id: $engagementId, active: true, owningOrgId: $owningOrgId})
+          ,(product:Product {active:true})
+        WITH COUNT(product) as total, engagement, product
+          RETURN total, product.id as id, product.createdAt as createdAt
+          ORDER BY ${sort} ${order}
+          SKIP $skip LIMIT $count
+      `;
```

Comment:
We need to compose the query using the functions from `query.helpers.ts`. The `organization` service's `list` function is a good example.